### DISCUSSION TITLE
Removing an invalid path from PATH

### DISCUSSION THREAD

**Discussion by SeSodesa** [fish-shell/fish-shell#9450] [Q&A] [answered]
I have a little problem with having added an apparently "invalid" path into the global `PATH` variable and now being unable to remove it because of the invalidity. To be more specific, here is the issue:
```fish
>> set --show PATH
$PATH: set in global scope, exported, a path variable with 14 elements
$PATH[1]: |/home/santtu/.cargo/bin|
$PATH[2]: |/home/santtu/.local/bin|
$PATH[3]: |/usr/local/sbin|
$PATH[4]: |/usr/local/bin|
$PATH[5]: |/usr/sbin|
$PATH[6]: |/usr/bin|
$PATH[7]: |/sbin|
$PATH[8]: |/bin|
$PATH[9]: |/usr/games|
$PATH[10]: |/usr/local/games|
$PATH[11]: |/usr/local/julia/1.7.2/bin|
$PATH[12]: |/usr/local/texlive/2022/bin/x86_64-linux|
$PATH[13]: |/snap/bin|
$PATH[14]: |/snap/bin|

>> set --erase --universal $PATH[11]
set: /usr/local/julia/1.7.2/bin: invalid variable name. See `help identifiers`

(Type 'help set' for related documentation)
```

I am guessing that it is the purely numeric folder name `1.7.2` in `$PATH[11]` that is causing this problem. My guess is that a similar issue might then occur with `$PATH[12]` as well.

What are my options regarding the removal of this particular entry? Should I just run something like
```fish
set -U PATH paths I want to maintain
```
or is there another alternative where `fish` just does not notice that there is something invalid in the thing it is trying to read and therefore just does what I ask it to?

**Comment by krobelus** [comment, marked as answer]
> set --erase --universal $PATH[11]

try

    set --erase --universal PATH[11]

**Reply by SeSodesa** [reply]
Ah, I think that was it. I needed to reference the list and not the thing it contains, of course.
